packages/java/src/node/java-contribution.ts

```diff
@@ -125,6 +142,18 @@ export class JavaContribution extends BaseLanguageServerContribution {
         });
     }
 
+    private generateDataFolderSuffix(workspaceUri: string): string {
+        const shaValue = sha1(workspaceUri);
+        let instanceCounter = 0;
+        let dataFolderName = shaValue + '_' + instanceCounter;
+        while (this.activeDataFolders.has(dataFolderName)) {
+            instanceCounter++;
+            dataFolderName = shaValue + '_' + instanceCounter;
+        }
+        this.activeDataFolders.add(dataFolderName);
```

Comment:
just a minor: I'd prefer to move the side effect to `start`.

looks good otherwise, let me check.